[12.0][MIG] base_custom_info: Migration to 12.0

Tardo opened the pull request:
Cc @tecnativa TT18593

carlosdauden committed:
[ADD][8.0] base_custom_info: Init develop

carlosdauden committed:
[IMP][8.0] base_custom_info: Actions and menus. Refactor strings Info to Property

carlosdauden committed:
[FIX][8.0] base_custom_info: Minor fix and security

yajo committed:
Fix many bugs and styling, and prepare for OCA.

yajo committed:
[9.0][MIG][base_custom_info] Migration.

Mostly no changes from 8.0 after all.

yajo committed:
[CHG] base_custom_info: Relicense to LGPL with all authors' consent

# Conflicts:
#	base_custom_info/__manifest__.py

yajo committed:
[IMP]base_custom_info: New types and access rules system.

* Now you can define properties types, and access rules are inherited from the model/record
  linked to the custom info record.
* Simplified version of computed value.
* Implement for res.partner.
* Add tests and fix bugs discovered in the meantime.
* Allow to disable partner custom info tab, and custom info menu.
* All of it can be set within general settings.
* Now, by default, this module does not display custom info for partners unless in demo mode.
  Better fit for a base module.
* You can disable the top menu entry too if it disturbs you, or enable it for everybody.
* Give a special form when editing in partner custom info tab.
* Sortable properties.
* Sort values at onchange time.
* Improve performance in onchange.
* Split in several model files.

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

pedrobaeza committed:
[MIG] Make modules uninstallable

sergio-teruel committed:
[FIX] base_custom_info: fix demo template csv

yajo committed:
[FIX][base_custom_info] Make tests work

fanha99 committed:
[MIG] [10.0] base custom info

lmignon committed:
[FIX] base_custom_info: Check contrain on model into template before the write
We must avoid to rely on the order in which computed fields (including related fields) and constrains methods are applied.
Due to a recent change into the ORM, the contrains on the model_id into CustomInfoTemplate is now called AFTER the recompute of the related model field into property_ids.info_value_ids
As side effect, when the constrains is called, the model on the info value is already updated with the new value and we no more know the old value....

OCA-git-bot committed:
base_custom_info 10.0.1.1.0

Tardo committed:
[MIG] base_custom_info: Migration to 12.0

AdriaGForgeFlow commented:
Are you planning on finishing this? Do you need help?

pedrobaeza commented:
Yes, we need to check a problem with the onchanges. I'll check this week.

AdriaGForgeFlow commented:
> Yes, we need to check a problem with the onchanges. I'll check this week.

Okay, thanks!

pedrobaeza committed:
[MIG][FIX] base_custom_info: Several adjustments:

- Put force_save=1 for allowing the web client to send readonly values to server
- Switch widget="selection" to "many2one" default with options because it's not
  supported anymore in tree view
- Move value_id to parent view.

pedrobaeza commented:
This should be now reviewable. @Tardo I think you can squash a bit commits.

Tardo committed:
[IMP] base_custom_info: Tests

Tardo commented:
Ready for review

pedrobaeza commented:
@AdriaGForgeFlow can you review?

AdriaGForgeFlow commented on base_custom_info/README.rst:
```diff
@@ -0,0 +1,288 @@
+
+The *value* will always be one of these.
+
+Recursive templates using options
```
Is this feature enabled? I don't seem to find how to use it.

Tardo replied:
I think that this refers to the feature of mix properties of different templates for a selection.

AdriaGForgeFlow replied:
Yes, but how do you use it? How do you specify in a selection property that links to a different template?
When I follow the instructions in the README and select "Needs videogames" in the Smart Partners template, nothing happens, no new properties appear in the list to be filled. Even though I see this in the property form view:
![image](https://user-images.githubusercontent.com/44768500/71069799-bb93b380-2179-11ea-9f6c-d5ccb1c07165.png)

See:
![image](https://user-images.githubusercontent.com/44768500/71069828-ccdcc000-2179-11ea-8845-3ff53891a190.png)

Tardo replied:
Finally, it's an outdated readme... moved to roadmap. This feature it's not implemented yet. Thanks!

AdriaGForgeFlow reviewed (changes requested):
Done a code review and functional review and overall looks good to me.

Just a small error when trying to access the Values menu in Custom Info / Basic.
![image](https://user-images.githubusercontent.com/44768500/70992065-25a05000-20c9-11ea-8d2f-b7688db10afd.png)

Another comment below.

Tardo commented:
Change done! thanks for the review

pedrobaeza commented:
Let's merge it then!

/ocabot merge
